src/core/socket.c

```diff
@@ -2058,6 +2107,21 @@ static void socket_enter_running(Socket *s, int cfd) {
                         return;
                 }
 
+                if (s->max_connections_per_source > 0) {
```

Comment:
our hash op functions for the hashmap cannot deal with address family that aren't AF_INET or AF_INET6. That's OK, but this means we should refuse the counting here for other address family. Otherwise we'll end up doing the tracking half-way for AF_UNIX for example, where all this would semi-break.